paddle/framework/op_registry.h

```diff
@@ -75,6 +81,20 @@ class OpRegistry {
   static std::unique_ptr<OperatorBase> CreateOp(const OpDesc& op_desc);
 
   static std::unique_ptr<OperatorBase> CreateGradOp(const OperatorBase& op);
+
+  // compile time InferShape
+  static void InferShape(const OpDesc& op_desc,
+                         std::map<std::string, VarDesc*>& var_descs) {
+    auto& info = OpInfoMap::Instance().Get(op_desc.type());
+    auto op = OpRegistry::CreateOp(op_desc);
```

Comment:
It is very strange that when checking `compile-time` infer shape, it actually creates a runtime operator.

Maybe add a `FIXME` comment here, I actually did not figure out what's the better way, but there should be a better way.

Reply:
removed now, Infershape is implemented inside Operator, so do not need to create a operator now